test/e2e-shared-tests/helpers.go

```diff
@@ -53,16 +53,36 @@ func waitForConfigDriftMonitorStart(t *testing.T, cs *framework.ClientSet, node
 
 	start := time.Now()
 
+	ctx := context.Background()
+
 	err := wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
 		mcdPod, err := helpers.MCDForNode(cs, &node)
 		require.Nil(t, err)
 
 		logs, err := cs.Pods(mcdPod.Namespace).GetLogs(mcdPod.Name, &corev1.PodLogOptions{
 			Container: "machine-config-daemon",
-		}).DoRaw(context.TODO())
+		}).DoRaw(ctx)
 		require.Nil(t, err)
 
-		return strings.Contains(string(logs), configDriftMonitorStartupMsg), nil
+		splitLogs := strings.Split(string(logs), "\n")
+
+		// Scan the logs from the bottom up, looking for either a shutdown or startup message.
+		for i := len(splitLogs) - 1; i >= 0; i-- {
```

Comment:
(I might be missing context here..)

Q: is there no other way to check if configdrifmonitor is running other than checking logs? Can we not use cdm.IsRunning() since that's the canonical way to test if it's up what we use in other tests and in the daemon itself.

Reply:
Hmm, if I understand correctly, this is from the perspective of the main testing pod which spun up a cluster, and is now interacting with it to e2e test. We're in the perspective of a user and not the actual MCD pod itself.

Or do you mean that we can debug into the MCD pod and check the running processes to see if the config drift monitor is running?

Reply:
@yuqi-zhang re: perspective, you're right. :+1: 

second point is more what I'm generally asking - is there any way to actually just check if the configdriftmonitor is running other than checking for a pod log that it started?  Reading the func comments it seems that this may run after mcd is done so we can't rely on mcd state but overall wondering if there is a more direct vs indirect way to check if it's running.

Reply:
ah ok, that settles that for now then. thanks @cheesesashimi